src/OpenLoco/src/Version.cpp

```diff
@@ -1,28 +1,76 @@
-#include "OpenLoco.h"
-
-#define NAME "OpenLoco"
+#include "Version.h"
+#include <string>
 
 namespace OpenLoco
 {
-    const char version[] = NAME ", "
-#ifdef OPENLOCO_VERSION_TAG
-        OPENLOCO_VERSION_TAG
-#else
-#error "OPENLOCO_VERSION_TAG Not set. Rerun CMake."
-#endif
-#if defined(OPENLOCO_BRANCH) || defined(OPENLOCO_COMMIT_SHA1_SHORT) || !defined(NDEBUG)
-                                " ("
-#if defined(OPENLOCO_BRANCH) && defined(OPENLOCO_COMMIT_SHA1_SHORT)
-        OPENLOCO_COMMIT_SHA1_SHORT " on " OPENLOCO_BRANCH
-#elif defined(OPENLOCO_COMMIT_SHA1_SHORT)
-        OPENLOCO_COMMIT_SHA1_SHORT
-#elif defined(OPENLOCO_BRANCH)
-        OPENLOCO_BRANCH
-#endif
-#ifndef NDEBUG
-                                ", DEBUG"
-#endif
-                                ")"
-#endif
+    static constexpr std::string kOpenLocoName = "OpenLoco";
```

Comment:
Our Ubuntu jammy CI is complaining that std::string isn't constexpr. This is true for the old version of gcc/clang that it ships.

While I don't like macro-heavy code either, I'm not convinced avoiding it is worth it, in this case. I'm honestly inclined to revert to 7f60817b057c756a1dc9bd1b76362806da90ed7b.